src/source/PeerConnection/Retransimitter.c

```diff
@@ -0,0 +1,134 @@
+#define LOG_CLASS "Retransmitter"
+
+#include "../Include_i.h"
+
+STATUS createRetransmitter(UINT32 seqNumListLen, UINT32 validIndexListLen, PRetransmitter *ppRetransmitter)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PRetransmitter pRetransmitter = MEMALLOC(SIZEOF(Retransmitter) + SIZEOF(UINT16) * seqNumListLen
+            + SIZEOF(UINT64) * validIndexListLen);
+    CHK(pRetransmitter != NULL, STATUS_NOT_ENOUGH_MEMORY);
+    pRetransmitter->sequenceNumberList = (PUINT16) (pRetransmitter + 1);
+    pRetransmitter->seqNumListLen = seqNumListLen;
+    pRetransmitter->validIndexList = (PUINT64) (pRetransmitter->sequenceNumberList + seqNumListLen);
+    pRetransmitter->validIndexListLen = validIndexListLen;
+
+CleanUp:
+    if (STATUS_FAILED(retStatus) && pRetransmitter != NULL) {
+        freeRetransmitter(&pRetransmitter);
+        pRetransmitter = NULL;
+    }
+    if (ppRetransmitter != NULL) {
+        *ppRetransmitter = pRetransmitter;
+    }
+    LEAVES();
+    return retStatus;
+}
+
+STATUS freeRetransmitter(PRetransmitter* ppRetransmitter)
+{
+    ENTERS();
+
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(ppRetransmitter != NULL, STATUS_NULL_ARG);
+    // freeRtpPacket is idempotent
+    CHK(*ppRetransmitter != NULL, retStatus);
+
+    MEMFREE(*ppRetransmitter);
```

Comment:
It's preferred to set the pointee to NULL. Just use SAFE_MEMFREE instead of these two